agent/agent_test.go

```diff
@@ -376,15 +376,16 @@ var attributeParams = agent.AgentConfigParams{
 	Paths: agent.Paths{
 		DataDir: "/data/dir",
 	},
-	Tag:               names.NewMachineTag("1"),
-	UpgradedToVersion: jujuversion.Current,
-	Password:          "sekrit",
-	CACert:            "ca cert",
-	APIAddresses:      []string{"localhost:1235"},
-	Nonce:             "a nonce",
-	Controller:        testing.ControllerTag,
-	Model:             testing.ModelTag,
-	JujuDBSnapChannel: "4.0/stable",
+	Tag:                      names.NewMachineTag("1"),
+	UpgradedToVersion:        jujuversion.Current,
+	Password:                 "sekrit",
+	CACert:                   "ca cert",
+	APIAddresses:             []string{"localhost:1235"},
+	Nonce:                    "a nonce",
+	Controller:               testing.ControllerTag,
+	Model:                    testing.ModelTag,
+	JujuDBSnapChannel:        "4.0/stable",
+	NonSyncedWritesToRaftLog: false,
```

Comment:
since this defaults to false, better to test with true

Reply:
It was set up as such to mimic the expected default value for a bootstrapped controller. Note that the test at https://github.com/juju/juju/pull/11646/commits/32c8614d4522091e94c96afc5f31c326211f98e1#diff-336a19a3907c67c768036d0fae6aa384R668-R678 checks that it works as expected in both states. If you think that it's better to swap this around let me know and I will amend the relevant commit on Monday.